packages/core/src/internal/base/button.base.ts

```diff
@@ -136,8 +142,15 @@ export class CdsBaseButton extends LitElement {
   }
 
   private updateButtonAttributes() {
-    this.isAnchor = !!this.anchor;
-    this.readonly = this.readonly || this.isAnchor;
+    this.containsAnchor = !!this.anchor;
+    this.isAnchor = this.parentElement?.tagName === 'A';
+
+    if (this.isAnchor && this.parentElement) {
```

Comment:
Do we have unit tests for all this?

Reply:
hmm it should, I think I goofed when I rebased onto master. I'll get them added back.